src/peer/negotiator.js

```diff
@@ -606,13 +620,50 @@ class Negotiator extends EventEmitter {
       this._pc.removeStream(localStreams[0]);
     }
 
-    this._replaceStreamCalled = true;
-
-    // a chance to trigger (and do nothing) on removeStream.
-    setTimeout(() => {
+    // HACK: For some reason FF59 doesn't work when Chrome 64 renegotiates after updating the stream.
+    // However, simply updating the localDescription updates the remote stream if the other browser is firefox.
+    // Chrome 64 probably uses replaceTrack-like functions internally.
+    if (this._remoteBrowser === 'firefox') {
       this._pc.addStream(newStream);
-      this._pc.onnegotiationneeded = origOnNegotiationNeeded;
-    });
+
+      // use setTimeout to trigger (and do nothing) on add/removeStream.
+      setTimeout(() => {
+        // update the localDescription with the new stream information (after getting to the right state)
+        let promise;
+        if (this._originator) {
+          promise = this._makeOfferSdp()
+            .then(offer => {
+              return this._pc.setLocalDescription(offer);
+            })
+            .then(() => {
+              return this._pc.setRemoteDescription(this._pc.remoteDescription);
+            });
+        } else {
+          promise = this._pc
+            .setRemoteDescription(this._pc.remoteDescription)
+            .then(() => {
+              return this._pc.createAnswer();
+            })
+            .then(answer => {
+              return this._pc.setLocalDescription(answer);
+            });
+        }
+        // restore onnegotiationneeded in case we need it later.
+        promise.finally(() => {
+          this._pc.onnegotiationneeded = origOnNegotiationNeeded;
+        });
+      });
+    } else {
+      // this is the normal flow where we renegotiate.
```

Comment:
Memo for @y-i and @leader22 : 
This `else` block is where Chrome M64's `_replacePerStream` goes through when the remote peer isn't Firefox.

Reply:
Should we comment it in source code?